desc/optimize/_constraint_wrappers.py

```diff
@@ -569,32 +493,76 @@ def build(self, eq=None, use_jit=None, verbose=1):  # noqa: C901
         for constraint in self._linear_constraints:
             constraint.build(self._eq, verbose=verbose)
 
+        self._objectives = combine_args(self._objective, self._constraint)
+        self._set_things(self._all_things)
+
         self._dim_f = self._objective.dim_f
         if self._dim_f == 1:
             self._scalar = True
         else:
             self._scalar = False
 
-        self.set_args()
-        self._objective.set_args(*self._full_args)
-        self._constraint.set_args(*self._full_args)
+        self._set_state_vector()
         # history and caching
-        self._x_old = np.zeros((self._dim_x,))
-        for arg in self.args:
-            self._x_old[self.x_idx[arg]] = getattr(eq, arg)
-
+        self._x_old = self.x(eq)
         self._allx = [self._x_old]
         self._allxopt = [self._objective.x(eq)]
         self._allxeq = [self._constraint.x(eq)]
-        self.history = {}
-        for arg in arg_order:
-            self.history[arg] = [np.asarray(getattr(self._eq, arg)).copy()]
+        self.history = [[self._eq.params_dict]]
```

Comment:
OK so before `history` was a dict of lists, and now it is a list of dicts?

Reply:
yeah. The outer list is iteration, the inner list is futureproofing for having multiple objects being optimized